### PR TITLE
test: Playwright を導入して E2E テスト雛形を追加

### LINKED ISSUE #91: test: Playwright を導入して E2E テスト追加
## 概要

issue #86 のステップ 5。最終的なユーザー操作レベルでの担保として Playwright による E2E テストを整備する。

## インストール

```bash
pnpm add -D playwright @playwright/test
```

## 設定ファイル

- [ ] `playwright.config.ts` — `baseURL`、認証セットアップ
- [ ] `package.json` の scripts に `test:e2e` を追加
- [ ] `.env.test` — テスト用 Supabase 接続情報（必要なら）

## チェックリスト

- [ ] ログイン〜動画登録までの一連フロー
- [ ] cast-selector で芸人を選択して保存
- [ ] メモの追加・編集（認証ユーザーのみ）
- [ ] 公開側で動画一覧が表示される

## 完了条件

- Playwright E2E はローカル実行できる状態
- CI 統合は別 issue でも可

## 関連

- 元 issue: #86
- 先行 PR: #87 (Vitest 環境整備 + Step 1)

### PR DESCRIPTION
**Pull request by hiiragi17**
## 概要

issue #86 のステップ 5。E2E テストの土台として Playwright を導入し、最低限のテスト雛形を追加した。

Closes #91

### セットアップ
- `playwright` と `@playwright/test` を devDependencies に追加
- `playwright.config.ts` を追加
  - `baseURL` は `PLAYWRIGHT_BASE_URL` で上書き可能（デフォルト `http://localhost:3000`）
  - `webServer` で `pnpm dev` を自動起動（既存サーバーがあれば再利用）
  - Chromium プロジェクトのみ、CI 時は retries=2 / workers=1 / reporter=github
- `package.json` に `test:e2e` / `test:e2e:ui` スクリプトを追加
- `.env.test.example` を追加（`.gitignore` に例外指定）
- `.gitignore` / eslint config に `playwright-report` / `test-results` を追加

### E2E spec（`e2e/`）

| ファイル | 内容 |
| --- | --- |
| `public.spec.ts` | トップページ表示、動画一覧の見出し表示 |
| `auth.spec.ts` | 未ログインで `/admin` → `/auth/login` リダイレクト、ログインフォームの required/type 検証 |
| `admin-flow.spec.ts` | `E2E_TEST_EMAIL` / `E2E_TEST_PASSWORD` が設定されたときのみ動くログイン後フロー。cast-selector とメモは `test.fixme` で TODO として残してある |

## 完了条件との対応

- [x] `playwright.config.ts`（`baseURL`、認証セットアップ用の env を整備）
- [x] `package.json` の scripts に `test:e2e` を追加
- [x] `.env.test`（`.env.test.example` として雛形を追加）
- [x] ローカルで `pnpm test:e2e` を実行できる状態
- [ ] cast-selector で芸人を選択して保存（`test.fixme` で TODO 化）
- [ ] メモの追加・編集（`test.fixme` で TODO 化）

issue 本文どおり「Playwright E2E はローカル実行できる状態 / CI 統合は別 issue でも可」を満たす範囲に留めた。シナリオの本実装は別途追加していく想定。

## Test plan

- [ ] `pnpm install` 後、`pnpm exec playwright install chromium` でブラウザを取得
- [ ] `pnpm dev` を別タブで起動した状態、または webServer 任せで `pnpm test:e2e` を実行し、`public.spec.ts` と `auth.spec.ts` がグリーンになることを確認
- [ ] `.env.test` に有効な `E2E_TEST_EMAIL` / `E2E_TEST_PASSWORD` を設定し、`admin-flow.spec.ts` のログイン後ページ表示が通ることを確認

---
_Generated by [Claude Code](https://claude.ai/code/session_01S4SyghvUxtLudwaDdkvMng)_

### CODE REVIEW

File: playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  projects: [
+    {
+      name: "chromium",
+      use: { ...devices["Desktop Chrome"] },
+    },
+  ],
+  webServer: {
+    command: "pnpm dev",
+    url: baseURL,
```
[the remark this answers is not in the document]
Comment (hiiragi17) [reply playwright.config.ts]: ae4f6a1 で対応しました。`baseURL` から URL を解析してポートを取り出し、`webServer.command` を `pnpm dev --port ${port}` にして起動コマンドと readiness URL のポートを揃えました。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: package.json [hunk 2]
```diff
@@ -8,7 +8,9 @@
     "start": "next start",
     "lint": "eslint",
     "test": "vitest run",
-    "test:watch": "vitest"
+    "test:watch": "vitest",
+    "test:e2e": "playwright test",
```
[the remark this answers is not in the document]
Comment (hiiragi17) [reply package.json:12]: ae4f6a1 で対応しました。`dotenv` を devDependency に追加し、`playwright.config.ts` の冒頭で `.env.test` が存在すれば自動的に読み込むようにしたので、`pnpm test:e2e` 経由でも `E2E_TEST_EMAIL` / `PLAYWRIGHT_BASE_URL` 等が反映されます。

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by hiiragi17** [comment]
@coderabbitai レビューして

**Comment by hiiragi17** [comment]
@coderabbitai レビューありがとうございます。d257668 で対応しました。

- 🔴 **playwright の二重インストール**: `devDependencies` から `playwright` を削除し `@playwright/test` のみに統一しました。`npx playwright test --list` で 7 件のテストが正しく認識されることを確認済みです。
- 🟡 **dotenv のログ出力**: `playwright.config.ts` の `loadEnv` 呼び出しに `quiet: true` を追加し、実行時ログを抑制しました。
- 🟡 **ポート解析の edge case**: 本 PR では issue #91 の通りローカル実行（`localhost`）専用の想定のため現状維持とします。ステージング等で `webServer` を無効化する分岐は将来の課題として認識しました。
- 🟢 **storageState による認証状態の再利用**: 妥当なご提案ですが、現状 `admin-flow.spec.ts` はログイン後フロー 1 件のみで `test.fixme` の TODO が中心のため、E2E シナリオを本実装する後続対応で導入を検討します。


---
_Generated by [Claude Code](https://claude.ai/code)_
